### PR TITLE
hardening/4_add_public_path_list

### PR DESCRIPTION
**Pull request by pcoello25**
- Fix issue #4 
- Testing changes: (Current public path list: `public` and `opendata`)

```
[ACCESS TO USER PATH]
curl -X GET "http://localhost:14000/webhdfs/v1/user/pcoello25?op=liststatus&user.name=pcoello25" -H "X-Auth-Token: xxxxxxxxxxxxxxxxxxxxxx"

**cosmos-proxy log messages**
info: Authentication OK: {"organizations": [], "displayName": "pcoello25", "roles": [], "app_id": "4d1af2eec3754099a4f8dc86bf735068", "isGravatarEnabled": false, "email": "pcv@xxx.com", "id": "pcoello25"}
info: Authorization OK: user pcoello25 is allowed to access /webhdfs/v1/user/pcoello25
info: Redirecting to http://<host>:<port>

[ACCESS TO A PUBLIC PATH]
curl -X GET "http://localhost:14000/webhdfs/v1/user/public?op=liststatus&user.name=pcoello25" -H "X-Auth-Token: xxxxxxxxxxxxxxxxxxxxxx"

**cosmos-proxy log messages**
info: Authentication OK: {"organizations": [], "displayName": "pcoello25", "roles": [], "app_id": "4d1af2eec3754099a4f8dc86bf735068", "isGravatarEnabled": false, "email": "pcv@xxx.com", "id": "pcoello25"}
info: Authorization OK: user pcoello25 is allowed to access /webhdfs/v1/user/public
info: Redirecting to http://<host>:<port>

[ACCESS TO A PRIVATE PATH]
curl -X GET "http://localhost:14000/webhdfs/v1/user/private?op=liststatus&user.name=pcoello25" -H "X-Auth-Token: 8uRbcGZ5Mkfz5yGojfUNGtALTCTpHV"
Authorization error: user pcoello25 cannot access /webhdfs/v1/user/private

**cosmos-proxy log messages**
info: Authentication OK: {"organizations": [], "displayName": "pcoello25", "roles": [], "app_id": "4d1af2eec3754099a4f8dc86bf735068", "isGravatarEnabled": false, "email": "pablo.coellovillalba@telefonica.com", "id": "pcoello25"}
error: Authorization error: user pcoello25 is not allowed to access /webhdfs/v1/user/private
```
- Assignee: @frbattid

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,13 +75,14 @@ To be done.
...
 
 * **host**: FQDN or IP address of the host running the proxy.
-* **port**: TCP listening port for incomming proxied requests.
+* **port**: TCP listening port for incoming proxied requests.
 * **target**:
     * **host**: FQDN or IP address of the host running the real service.
     * **port**: TCP listening port of the real service.
 * **idm**:
     * **host**: FQDN or IP address where the Identity Manager runs. Do not write it in URL form!
     * **port**: Port where the Identity Manager listens for requests. Typically 443.
```
Comment (frbattid) [inline README.md]: Port --> port

File: README.md [hunk 2]
```diff
@@ -75,13 +75,14 @@ To be done.
...
+* **public_path_list**: list of public path can be reached for all users.
```
Comment (frbattid) [inline README.md]: path --> paths that can be reached by all users.

File: conf/cosmos-proxy.json [hunk 3]
```diff
@@ -9,6 +9,10 @@
     "host": "account.lab.fiware.org",
     "port": "443"
   },
+  "public_path_list" : [
```
Comment (frbattid) [inline conf/cosmos-proxy.json]: public_paths_list

File: src/cosmos-proxy.js [hunk 4]
```diff
@@ -49,16 +59,24 @@ http.createServer(function (req, res) {
                 res.end('Authentication error: ' + result);
             } else {
                 logger.info('Authentication OK: ' + result);
+                var whiteListed = isWhiteListed(conf.public_path_list,path);
```
Comment (frbattid) [inline src/cosmos-proxy.js]: public_paths_list

File: README.md (hunk 1 above)
Comment (pcoello25) [inline README.md]: Fixed in 1347771

File: README.md (hunk 2 above)
Comment (pcoello25) [inline README.md]: Fixed in 1347771

File: conf/cosmos-proxy.json (hunk 3 above)
Comment (pcoello25) [inline conf/cosmos-proxy.json]: Fixed in 8840fff

File: src/cosmos-proxy.js (hunk 4 above)
Comment (pcoello25) [inline src/cosmos-proxy.js]: Fixed in 8840fff

**Comment by frbattid** [comment]
LGTM

**Pull request merged**